packages/matter.js/src/cluster/GeneralDiagnosticCluster.ts

```diff
@@ -0,0 +1,209 @@
+
+    /** list of the unicast IPv6 addresses that are currently assigned to the network interface. */
+    iPv6Addresse: TlvField(5, TlvArray(TlvString.bound({ maxLength: 8 }))),
+
+    /** indicate the type of the interface */
+    type: TlvField(6, TlvEnum<InterfaceType>()),
+});
+
+/** SHALL indicate a change in the set of hardware faults currently detected by the Node
+   * @see {@link MatterCoreSpecificationV1_0} §11.11.9.1
+ */
+const HardwareFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 11 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 11 })),
+};
+
+/** SHALL indicate a change in the set of radio faults currently detectedby the Node
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.9.2
+ */
+const RadioFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 7 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 7 })),
+};
+
+/** SHALL indicate a change in the set of network faults currently detected by the Node
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.9.3
+*/
+const NetworkFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 4 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 4 })),
+};
+
+/** SHALL indicate the reason that caused the device to start-up
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.7.4  
+*/
+const BootReasonEvent = {
+    bootReason: TlvField(0, TlvEnum<BootReason>()),
+};
+
+/** Provide a means for certification tests to trigger some test-plan-specific events 
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.8.1 
+*/
+const TlvTestEventTriggerRequest = TlvObject({
+    enableKey: TlvField(0, TlvString.bound({ length: 16 })),
+    eventTrigger: TlvField(0, TlvUInt64)
+});
+
+/**
+ * The General Diagnostics Cluster, along with other diagnostics clusters, provide a means to acquire
+ * standardized diagnostics metrics that MAY be used by a Node to assist a user or Administrator in
+ * diagnosing potential problems. The General Diagnostics Cluster attempts to centralize all metrics
+*  that are broadly relevant to the majority of Nodes.
+ * 
+ * @see {@link MatterCoreSpecificationV1_0} § 11.11
+ */
+export const GeneralDiagnosticCluster = Cluster({
+    id: 0x033,
+    name: "GeneralDiagnostic",
+    revision: 1,
+    features: {
+        /** The ability to store a name for a group. */
+        groupNames: BitFlag(0),
+    },
+
+    /** @see {@link MatterCoreSpecificationV1_0} § 11.11.7 */
+    attributes: {
+
+        /** list of NetworkInterface structs */
+        networkInterfaces: Attribute(0, TlvArray(TlvNetworkInterface, { length: 254 })),
+
+        /**   SHALL indicate a best-effort count of the number of times the Node has rebooted. */
+        rebootCount: Attribute(1, TlvUInt16, { default: 0 }),
```

Comment:
Please add "persistent: true" because of the "N" in Quality column

Reply:
persistent is a valid "AttributeOptions" in node-matter, but not in matter.js - waiting for a PR for cluster.ts?

Reply:
I handle it